### PR TITLE
Test calculateRewards and add claimRewards

### PR DESCRIPTION
**Pull request by waynehoover**
Small addition to test `calculateRewards`. Probably need more to test for unhappy paths too.

### CODE REVIEW

File: contracts/Straddle.sol [hunk 1]
```diff
@@ -158,11 +158,11 @@ contract Straddle is Context, Ownable, ERC20("Straddle", "STRAD") {
...
         uint deposited = getUserDepositBalance(msg.sender);
         uint locked = getUserLockedBalance(msg.sender);
-        require(deposited > 0, "No funds to withdraw");
+        require(deposited > 0, "No STRAD available to withdraw");
 
         uint availableToWithdraw = deposited - locked;
-        require(availableToWithdraw > 0, "No unlocked funds available to withdraw");
-        require(availableToWithdraw >= amount, "Amount to withdraw exceeds available funds");
+        require(availableToWithdraw > 0, "No unlocked STRAD available to withdraw");
+        require(availableToWithdraw >= amount, "Amount to withdraw exceeds available STRAD");
```
Comment (tfire) [inline contracts/Straddle.sol:165]: good call

**Comment by tfire** [comment]
So it appears we were both working on this at 11pm on a Saturday - cheers 🥂 

Bit of a merge conflict but it shouldn't be hard to resolve. 

I was talking to Bazz and he wanted to go with a staging pool model for the USDC.

What I mean by staging pool is that the `distribute(uint)` becomes 

```
function depositRewards(uint usdc)
function distributeRewards() 
```

so in order to fix the conflict, what was previously, just add the deposit call in front of the distribute call.

**Comment by tfire** [comment]
Also I think the build will succeed if I invite you to the repository formally speaking

**Comment by tfire** [comment]
Just a personal nitpick but I would prefer to leave the sample files around to look at

**Comment by waynehoover** [comment]
going to open this up on the repo instead of fork.

**Pull request closed without merging**